domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonTest.kt

```diff
@@ -0,0 +1,164 @@
+
+private const val BOOLEAN_PLATFORM_PARAMETER_NAME = "boolean_platform_parameter_name"
+private const val BOOLEAN_PLATFORM_PARAMETER_VALUE = true
+
+private const val INCORRECT_PLATFORM_PARAMETER_NAME = "incorrect_platform_parameter_name"
+
+/**
+ * [PlatformParameterSingletonTest] verifies the working of [PlatformParameterSingleton] by testing
+ * the [PlatformParameterValue] received in different cases
+ */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterSingletonTest.TestApplication::class)
+class PlatformParameterSingletonTest {
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  private val mockPlatformParameterMap by lazy {
+    val stringPlatformParameter = PlatformParameter.newBuilder()
+      .setString(STRING_PLATFORM_PARAMETER_VALUE).build()
+    val integerPlatformParameter = PlatformParameter.newBuilder()
+      .setInteger(INTEGER_PLATFORM_PARAMETER_VALUE).build()
+    val booleanPlatformParameter = PlatformParameter.newBuilder()
+      .setBoolean(BOOLEAN_PLATFORM_PARAMETER_VALUE).build()
+
+    mapOf<String, PlatformParameter>(
+      STRING_PLATFORM_PARAMETER_NAME to stringPlatformParameter,
+      INTEGER_PLATFORM_PARAMETER_NAME to integerPlatformParameter,
+      BOOLEAN_PLATFORM_PARAMETER_NAME to booleanPlatformParameter,
+    )
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_isEmpty() {
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isEmpty()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_isNotEmpty() {
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isEmpty()
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isNotEmpty()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMapTwice_checkIsNotUpdatedTwice() {
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isNotEmpty()
+
+    val emptyPlatformParameterMap = mapOf<String, PlatformParameter>()
+    platformParameterSingleton.setPlatformParameterMap(emptyPlatformParameterMap)
+
+    assertThat(
+      platformParameterSingleton.getPlatformParameterMap()
+    ).isNotEqualTo(emptyPlatformParameterMap)
+    assertThat(
+      platformParameterSingleton.getPlatformParameterMap()
+    ).isEqualTo(mockPlatformParameterMap)
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_retrieveStringParameter_verifyItsValue() {
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    val stringPlatformParameter = platformParameterSingleton.getStringPlatformParameter(
+      STRING_PLATFORM_PARAMETER_NAME
+    )
+    assertThat(stringPlatformParameter?.value).isEqualTo(STRING_PLATFORM_PARAMETER_VALUE)
```

Comment:
ah, I missed this earlier, is passing a nullable in `assertThat` is a good idea? 

What happens if we pass a null value to `assertThat`? 
Also, does the testing goes correctly if passing a null value?
Or can we modify the test something like where we confirm that we will get the desired value and null never shows up?

Reply:
@anandwana001 This nullable result is part of our logic.
We pass a name to the gettter method (inside singleton) then this getter method looks for the parameter corresponding to this name in the map. If it find this map then it returns that PlatformParameter instance but if in case there is no such parameter in the map then in that case this getter method returns null. Now it is the responsibility of the dagger module to provide a default PlatformParameter whenever singleton returns null. 

So for our test -> We are checking that after passing in the correct name (means name of a parameter which exists in map) do we have the required value. This required value is not going to be null and should be equal to true value (which we set at the start of the test). 

Hence this single assertThat covers a test for both cases ie it checks that the returned value should not be null and it should be equal to the true value.